debezium-connector-oracle/src/test/java/io/debezium/connector/oracle/OracleConnectorIT.java

```diff
@@ -2961,6 +2965,174 @@ private void consumeRecords(Configuration config) throws SQLException, Interrupt
         stopConnector();
     }
 
+    @Test
+    @FixFor("DBZ-4367")
+    public void shouldCaptureChangesForTransactionsAcrossSnapshotBoundary() throws Exception {
+        TestHelper.dropTable(connection, "DBZ4367");
+        try {
+            connection.execute("CREATE TABLE DBZ4367 (ID number(9, 0), DATA varchar2(50))");
+            TestHelper.streamTable(connection, "DBZ4367");
+
+            connection.execute("INSERT INTO DBZ4367 (ID, DATA) VALUES (1, 'pre-snapshot pre TX')");
+            connection.executeWithoutCommitting("INSERT INTO DBZ4367 (ID, DATA) VALUES (2, 'pre-snapshot in TX')");
+
+            Configuration config = TestHelper.defaultConfig()
+                    .with(OracleConnectorConfig.TABLE_INCLUDE_LIST, "DEBEZIUM\\.DBZ4367")
+                    .build();
+            start(OracleConnector.class, config);
+            waitForStreamingRunning(TestHelper.CONNECTOR_NAME, TestHelper.SERVER_NAME);
+
+            connection.executeWithoutCommitting("INSERT INTO DBZ4367 (ID, DATA) VALUES (3, 'post-snapshot in TX')");
+            connection.executeWithoutCommitting("COMMIT");
+            connection.execute("INSERT INTO DBZ4367 (ID, DATA) VALUES (4, 'post snapshot post TX')");
+
+            SourceRecords records = consumeRecordsByTopic(4);
+            List<Integer> ids = records.recordsForTopic("server1.DEBEZIUM.DBZ4367").stream()
+                    .map(r -> getAfter(r).getInt32("ID"))
+                    .collect(Collectors.toList());
+            assertThat(ids).containsOnly(1, 2, 3, 4);
+            assertThat(ids).doesNotHaveDuplicates();
+            assertThat(ids).hasSize(4);
+        }
+        finally {
+            stopConnector();
+            TestHelper.dropTable(connection, "DBZ4367");
+        }
+    }
+
+    @Test
+    @FixFor("DBZ-4367")
+    public void shouldCaptureChangesForTransactionsAcrossSnapshotBoundaryWithoutDuplicatingSnapshottedChanges() throws Exception {
```

Comment:
I ran this test directly and the test passed but if I run the test suite it fails:
```
[ERROR] Failures: 
[ERROR]   OracleConnectorIT.shouldCaptureChangesForTransactionsAcrossSnapshotBoundaryWithoutDuplicatingSnapshottedChanges:3026 <[1, 2]> does not contain element(s):<[3]>
[INFO] 
```
Any idea why?

Reply:
This is really a tough issue to crack @chanetd because I can reproduce it by running all the tests locally in my IDE inside the `OracleConnectorIT` class but if I specifically target this test by itself or set this test to run on repeat that's when I cannot reproduce it.  The issue is that the record with `ID=2` gets emitted in the snapshot and should not.  It's not clear in the test how that would even be possible.

Reply:
So the issue is one where another test explicitly enables auto-commit and never resets it after the test concludes which leads to tainted behavior in the test you introduced.